var/spack/repos/builtin/packages/gcc-runtime/package.py

```
@@ -0,0 +1,151 @@
+
+    def _get_libraries_elf(self):
+        """Get the GCC runtime libraries for ELF binaries"""
+        cc = Executable(self.compiler.cc)
+        lib_regex = re.compile(rb"\blib[a-z-_]+\.so\.\d+\b")
+        path_and_install_name = []
+
+        for name in self.LIBRARIES:
+            # Look for the dynamic library that gcc would use to link,
+            # that is with .so extension and without abi suffix.
+            path = cc(f"-print-file-name=lib{name}.so", output=str).strip()
+
+            # gcc reports an absolute path on success
+            if not os.path.isabs(path):
+                continue
+
+            # Now there are two options:
+            # 1. the file is an ELF file
+            # 2. the file is a linker script referencing the actual library
+            with open(path, "rb") as f:
+                try:
+                    # Try to parse as an ELF file
+                    soname = parse_elf(f, dynamic_section=True).dt_soname_str.decode("utf-8")
+                except Exception:
+                    # On failure try to "parse" as ld script; the actual
+                    # library needs to be mentioned by filename.
+                    f.seek(0)
+                    script_matches = lib_regex.findall(f.read())
+                    if len(script_matches) != 1:
+                        continue
+                    soname = script_matches[0].decode("utf-8")
+
+            # Now locate and install the runtime library
+            runtime_path = cc(f"-print-file-name={soname}", output=str).strip()
+
+            if not os.path.isabs(runtime_path):
+                continue
+
+            path_and_install_name.append((runtime_path, soname))
+
+        return path_and_install_name
+
+    def _get_libraries_macho(self):
+        """Same as _get_libraries_elf but for Mach-O binaries"""
+        cc = Executable(self.compiler.cc)
+        path_and_install_name = []
+
+        for name in self.LIBRARIES:
```

same question here as above -- this should fail fast if it doesn't find some reasonable set of libs.

---

See above